### PR TITLE
Add tests for reverse PTR records

### PR DESCRIPTION
**Pull request by href**
This includes a test for Floating IPs (v4/v6), and servers.

### CODE REVIEW

File: test_public_network.py [hunk 1]
```diff
@@ -153,3 +155,58 @@ def test_public_network_ipv4_only_on_all_images(prober, create_server, image):
...
+
+    # Create a Floating IP and wait for propagation
+    ptr = f"at-{secrets.token_hex(4)}.example.org"
+    fip = create_floating_ip(
+        ip_version=ip_version, region=region, reverse_ptr=ptr)
+
+    def assert_ptr_propagated():
+        assert reverse_ptr(fip, 'ns1.cloudscale.ch') == f'{ptr}.'
+        assert reverse_ptr(fip, 'ns2.cloudscale.ch') == f'{ptr}.'
+        assert reverse_ptr(fip, 'ns3.cloudscale.ch') == f'{ptr}.'
```
Comment (gaudenz) [inline test_public_network.py:200]: Same as above.

**Review by gaudenz — CHANGES_REQUESTED** [review]
Looks good in general. Just the change to not hardcode the list of nameservers here.

File: util.py [hunk 2]
```diff
@@ -527,3 +530,17 @@ def is_port_online(host, port, timeout=1.0):
...
+def reverse_ptr(address, ns):
+    """ Queries the given nameserver for the PTR record of an IP. """
+
+    resolver = Resolver(configure=False)
+    resolver.nameservers.append(socket.gethostbyname(ns))
+
+    reverse = reversename.from_address(str(address))
+
+    try:
+        return resolver.resolve(reverse, 'PTR').rrset[0].to_text()
```
[the remark this answers is not in the document]
Comment (href) [reply util.py]: I'm pretty indifferent, so I went with your suggestion.

File: test_public_network.py [hunk 3]
```diff
@@ -153,3 +155,58 @@ def test_public_network_ipv4_only_on_all_images(prober, create_server, image):
...
+    ipv4 = server.ip('public', 4)
+    ipv6 = server.ip('public', 6)
+
+    def assert_ptr_propagated():
+        assert reverse_ptr(ipv4, 'ns1.cloudscale.ch') == f'{server.name}.'
+        assert reverse_ptr(ipv4, 'ns2.cloudscale.ch') == f'{server.name}.'
+        assert reverse_ptr(ipv4, 'ns3.cloudscale.ch') == f'{server.name}.'
+        assert reverse_ptr(ipv6, 'ns1.cloudscale.ch') == f'{server.name}.'
+        assert reverse_ptr(ipv6, 'ns2.cloudscale.ch') == f'{server.name}.'
+        assert reverse_ptr(ipv6, 'ns3.cloudscale.ch') == f'{server.name}.'
```
[the remark this answers is not in the document]
Comment (href) [reply test_public_network.py]: I implemented an approach that is more generic. I would have liked to surface the DNS servers we use in the test, but after trying it out I didn't feel like hiding them is a big loss.

**Review by gaudenz — APPROVED** [review]
LGTM now. I suggest to squash the commits before merging.

**Pull request merged**